libs/core/src/lib/slider/slider.component.ts

```diff
@@ -0,0 +1,748 @@
+        const { x, width } = this.trackEl.nativeElement.getBoundingClientRect();
+        let percentage = (event.clientX - x) / width;
+
+        if (this._isRtl) {
+            percentage = 1 - percentage;
+        }
+
+        const newValue = this.min + percentage * (this.max - this.min);
+
+        return this._processNewValue(newValue, takeCustomValue);
+    }
+
+    /** @hidden */
+    private _processNewValue(newValue: number, takeCustomValue = true): number | SliderTickMark {
+        if (newValue > this.max) {
+            newValue = this.max;
+        }
+
+        if (newValue < this.min) {
+            newValue = this.min;
+        }
+
+        const stepDiffArray = this._valuesBySteps
+            .map((stepValue) => ({ diff: Math.abs(stepValue - newValue), value: stepValue }))
+            .sort((a, b) => a.diff - b.diff);
+        let value: SliderTickMark | number = stepDiffArray[0].value;
+
+        if (takeCustomValue && this.customValues.length > 0) {
+            value = this.customValues[value];
+        }
+
+        return value;
+    }
+
+    /** @hidden */
+    private _setRangeHandleValueAndPosition(handleIndex: RangeHandles, value: number): void {
+        const position = this._calcProgress(value, true);
+
+        if (handleIndex === RangeHandles.First) {
+            this._handle1Value = value;
+            this._handle1Position = position;
+        }
+
+        if (handleIndex === RangeHandles.Second) {
+            this._handle2Value = value;
+            this._handle2Position = position;
+        }
+
+        this._rangeProgress = Math.abs(this._handle2Position - this._handle1Position);
+    }
+
+    /** @hidden */
+    private _constructRangeModelValue(): number[] | SliderTickMark[] {
+        let rangeValue: number[] | SliderTickMark[] = [
+            Math.min(this._handle1Value, this._handle2Value),
+            Math.max(this._handle1Value, this._handle2Value)
+        ];
+
+        if (this.customValues.length > 0) {
+            const min = this.customValues[rangeValue[0]] || this.customValues[0];
+            const max = this.customValues[rangeValue[1]] || this.customValues[this.customValues.length - 1];
+
+            rangeValue = [min, max];
+        }
+
+        return rangeValue;
+    }
+
+    /** @hidden */
+    private _attachResizeListener(): void {
+        fromEvent(window, 'resize')
+            .pipe(debounceTime(500), takeUntil(this._onDestroy$))
+            .subscribe(() => this._onResize());
+    }
+
+    /** @hidden */
+    private _onResize(): void {
+        this._constructTickMarks();
+        this._cdr.detectChanges();
+    }
+
+    /** @hidden */
+    private _constructValuesBySteps(): void {
+        try {
+            this._valuesBySteps = Array((this.max - this.min) / this.step + 1)
+                .fill({})
+                .map((_, i) => Number((this.min + i * this.step).toFixed(2)));
```

Comment:
```suggestion
            this._valuesBySteps = [...Array((this.max - this.min) / this.step + 1).keys()]
                .map(i => Number((this.min + i * this.step).toFixed(2)));
```

Reply:
Will be error
`core.js:4442 ERROR RangeError: Invalid array length`

Reply:
changed to
```typescript
            this._valuesBySteps = Array.from({ length: (this.max - this.min) / this.step + 1 }, (_, i) =>
                Number((this.min + i * this.step).toFixed(2))
            );
```

Reply:
Have you pushed changes? I don't see any